ros2test/ros2test/command/test.py

```diff
@@ -12,21 +12,46 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import logging
+import os
+
 import launch_testing.launch_test
 import launch_testing_ros
 
 from ros2cli.command import CommandExtension
 
+from ..api.domain_coordinator import get_coordinated_domain_id
+
+_logger_ = logging.getLogger(__name__)
+
 
 class TestCommand(CommandExtension):
     """Run a ROS2 launch test."""
 
     def add_arguments(self, parser, cli_name):
         """Add arguments to argparse."""
         launch_testing.launch_test.add_arguments(parser)
+        parser.add_argument(
+            '--disable-isolation', action='store_true', default=False,
+            help='Disable automatic ROS_DOMAIN_ID isolation.'
+        )
 
     def main(self, *, parser, args):
         """Entry point for CLI program."""
+        # If ROS_DOMAIN_ID is already set, launch_test will respect that domain ID and use it.
+        # If ROS_DOMAIN_ID is not set, launch_test will pick a ROS_DOMAIN_ID that's not being used
+        # by another launch_test process.
+        # This is to allow launch_test to run in parallel and not have ROS cross-talk.
+        # If the user needs to debug a test and they don't have ROS_DOMAIN_ID set in their
+        # environment they can disable isolation by passing the --disable-isolation flag.
```

Comment:
@ivanpauno I'd try to condense much of these comments in the associated argument help string above.